compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/Utils.java

```diff
@@ -124,7 +124,10 @@ public static TypeSymbol getEffectiveType(IntersectionTypeSymbol intersectionTyp
     public static ObjectTypeSymbol getObjectTypeSymbol(Symbol serviceObjectTypeOrClass) {
         if (serviceObjectTypeOrClass.kind() == SymbolKind.TYPE_DEFINITION) {
             TypeDefinitionSymbol serviceObjectTypeSymbol = (TypeDefinitionSymbol) serviceObjectTypeOrClass;
-            return (ObjectTypeSymbol) serviceObjectTypeSymbol.typeDescriptor();
+            TypeSymbol typeSymbol = serviceObjectTypeSymbol.typeDescriptor();
+            if (typeSymbol.typeKind() == TypeDescKind.OBJECT) {
+                return (ObjectTypeSymbol) typeSymbol;
+            }
```

Comment:
Let's add an error message in the `UnsupportedOperationException` to make it easier to track the error if we ever face this (Hope not) exception.

Reply:
Error message added